components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -387,6 +389,25 @@ private OAuthConsumerAppDTO createOAuthApp(ApplicationRegistrationRequest regist
         oAuthConsumerApp.setTokenType(registrationRequest.getTokenType());
         oAuthConsumerApp.setBackChannelLogoutUrl(
                 validateBackchannelLogoutURI(registrationRequest.getBackchannelLogoutUri()));
+        if (CollectionUtils.isNotEmpty(registrationRequest.getAudiences())) {
+            OAuth2ServiceComponentHolder.setAudienceEnabled(true);
+            oAuthConsumerApp.setAudiences(registrationRequest.getAudiences().toArray(new String[0]));
+        }
+
+        if (registrationRequest.getIdTokenEncryptionAlgorithm() != null) {
+            oAuthConsumerApp.setIdTokenEncryptionEnabled(true);
+            oAuthConsumerApp.setIdTokenEncryptionAlgorithm(registrationRequest.getIdTokenEncryptionAlgorithm());
+            if (registrationRequest.getIdTokenEncryptionMethod() != null) {
+                oAuthConsumerApp.setIdTokenEncryptionMethod(registrationRequest.getIdTokenEncryptionMethod());
+            } else {
+                oAuthConsumerApp.setIdTokenEncryptionMethod("A128CBC-HS256");
```

Comment:
isn't there a util method to get default id token encryption method e.g. from 
OAuth2Utl?

Reply:
There is a method to get default value but it's set to "A128GCM"
[https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java#L186](url)

Reply:
we will need to fix it and use that because encryption method default value is configurable from identity.xml